### PR TITLE
Appeals | Fix date formatting

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > The `va-memorable-date` web component has 3 inputs that allow the Veteran to type in the year, month and day digits separately. When testing this component in [storybook](https://design.va.gov/storybook/?path=/docs/components-va-memorable-date--default) blurring the component after entering single digit month & day will add leading zeros. Within our appeals forms, the blur event doesn't always add these leading zeros. To prevent problems, we test and update these dates to ensure our forms don't get rejected.
  >
  > The date that triggered this work was `'2023-01-1 '` - which is at the expected 10 character length, but instead of a leading zero, there is a trailing space. Updates to the code trim the date string and re-format the date.
- _(If bug, how to reproduce)_
  > Unknown. The web component does not allow typing in non-digit characters
- _(What is the solution, why is this the solution)_
  > Trim date string and process & reformat the date string
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits decision reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

[#60051](https://github.com/department-of-veterans-affairs/va.gov-team/issues/60051)

## Testing done

- _Describe what the old behavior was prior to the change_
  > Added tests to ensure all whitespace is removed prior to processing a date string
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
  > Added new unit tests to ensure whitespace is removed from dates
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

N/A

## What areas of the site does it impact?

Appeals forms: HLR, NOD & Supplemental Claims

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/appeals/10182/tests/utils/replace.unit.spec.js [hunk 1]
```diff
@@ -50,6 +50,10 @@ describe('replaceSubmittedData', () => {
 });
 
 describe('fixDateFormat', () => {
+  it('should return invalid dates strings', () => {
+    expect(fixDateFormat()).to.eq('-00-00');
+    expect(fixDateFormat('200')).to.eq('200-00-00');
```
Comment (rocio-desantiago) [inline src/applications/appeals/10182/tests/utils/replace.unit.spec.js:55]: Should the 200 here be a year?

Comment (rocio-desantiago) [reply src/applications/appeals/10182/tests/utils/replace.unit.spec.js:55]: I think I'm missing why we're returning invalid date strings on purpose.

File: src/applications/appeals/10182/tests/utils/replace.unit.spec.js [hunk 2]
```diff
@@ -60,4 +64,11 @@ describe('fixDateFormat', () => {
     expect(fixDateFormat('2023-10-1')).to.eq('2023-10-01');
     expect(fixDateFormat('2000-6-30')).to.eq('2000-06-30');
   });
+  it('should return properly formatted date string when passed dates with weird spacing', () => {
+    expect(fixDateFormat('2020--')).to.eq('2020-00-00');
```
Comment (rocio-desantiago) [inline src/applications/appeals/10182/tests/utils/replace.unit.spec.js:68]: if `2020-00-0`0 were to return would that still be a valid date?

File: src/applications/appeals/10182/tests/utils/replace.unit.spec.js [hunk 3]
```diff
@@ -60,4 +64,11 @@ describe('fixDateFormat', () => {
...
+    expect(fixDateFormat('2020-1-')).to.eq('2020-01-00');
```
Comment (rocio-desantiago) [inline src/applications/appeals/10182/tests/utils/replace.unit.spec.js:69]: same question here. Would a `00` for the day still return a valid date?

**Review by rocio-desantiago — APPROVED** [review]

**Review by rocio-desantiago — COMMENTED** [review]
Questions about date formats

File: src/applications/appeals/10182/tests/utils/replace.unit.spec.js (hunk 3 above)
Comment (Mottie) [reply src/applications/appeals/10182/tests/utils/replace.unit.spec.js:69]: It's not a valid date. We want it to return `00` so the validation error kicks in - if we forced this to be `01`, the resulting  submitted date may not match what the Veteran is trying to enter.

**Pull request merged**